client/state/happychat/middleware.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import { ROUTE_SET } from 'state/action-types';
+import { sendEventMessage } from 'state/happychat/actions';
+import {
+	isHappychatClientConnected,
+	isHappychatAcceptingChats,
+	isHappychatChatAssigned
+ } from 'state/happychat/selectors';
+import { getCurrentUser } from 'state/current-user/selectors';
+
+export default ( store ) => ( next ) => ( action ) => {
+	if ( action.type === ROUTE_SET ) {
+		const state = store.getState();
+		const currentUser = getCurrentUser( store.getState() );
+		if ( isHappychatClientConnected( state ) &&
+			isHappychatAcceptingChats( state ) &&
+			isHappychatChatAssigned( state ) ) {
+			store.dispatch( sendEventMessage( `Looking at https://wordpress.com${ action.path }?support_user=${ currentUser.username } ` ) );
```

Comment:
I've left a comment in `actions.js` in the [other pending middleware PR](https://github.com/Automattic/wp-calypso/pull/12218/files) to this effect, all new logic that interacts with `connection` should go in middleware.

While working on #12218 I found that it was difficult to have a reference to `connection` in the middleware, since it's instantiated in `actions.js`. To solve that during the transition, I introduced `common.js` as a temporary measure.